tests/extend.test.ts

```diff
@@ -19,37 +23,77 @@ describe('Extend', () => {
 
   describe('contract owner', () => {
     let owner: JWKInterface;
+    let walletAddress: string;
+    let emptyWalletCaller: JWKInterface;
 
     beforeAll(async () => {
       owner = getLocalWallet(0);
+      walletAddress = await arweave.wallets.getAddress(owner);
       contract = warp.pst(srcContractId).connect(owner);
+      emptyWalletCaller = getLocalWallet(12);
     });
 
-    it('should not be able to extend a record that is not in its grace period', async () => {
-      const extendYears = MAX_YEARS - 1;
+    it('should not be able to extend a record if the caller has insufficient balance', async () => {
+      const extendYears = 1;
       const name = 'name1';
-      const { cachedValue: prevCachedValue } = await contract.readState();
+      const tempContract = warp.pst(srcContractId).connect(emptyWalletCaller);
```

Comment:
why `tempContract`?

Reply:
because that instance is for that specific test, not the entire suite

Reply:
you don't need to do `warp.pst(strContractId)` again - believe you can do `contract.connect(emptyWalletCaller)` - lmk if i'm wrong

but if you do that, we should add an `afterEach` that reconnects the owner wallet

Reply:
not a big deal

Reply:
huh, it is possible to do that, i assumed it wasnt! go-go gadget warp sdk haha.

Reply:
updated 👍